proxy/tracing.go

```diff
@@ -112,26 +145,61 @@ func (t *proxyTracing) startFilterTracing(operation string, ctx *context) *filte
 	if t.disableFilterSpans {
 		return nil
 	}
-	span := tracing.CreateSpan(operation, ctx.request.Context(), t.tracer)
+	span := &tracing.SpanWrapper{}
+
+	if t.useOtel {
+		var reqCtx goCtx.Context
+		reqCtx, span.Otel = t.otelTracer.Start(ctx.request.Context(), operation)
+		ctx.request.WithContext(reqCtx)
+		return &filterTracing{span, t.logFilterLifecycleEvents, t.useOtel}
+	}
+
+	span.Ot = tracing.CreateSpan(operation, ctx.request.Context(), t.otTracer)
```

Comment:
you can also implement `(*tracing.SpanWrapper).startFilterTracing(operation string, ctx *context)`